0000-stream/0000-stream.md

```diff
@@ -0,0 +1,174 @@
+- The server's ILP address
+
+The Server MAY generate the shared secret from a longer-term secret and a nonce, as described in [PSKv1's Appendix A: Recommended Algorithm for Generating Shared Secrets](./0016-pre-shared-key/0016-pre-shared-key.md#appendix-a-recommended-algorithm-for-generating-shared-secrets). In this case, the Server would append the nonce (**not** the secret) to their ILP address to create the destination address they communicate to the Client.
+
+### Multiplexed Money Streams
+
+STREAM accounts for value sent over a Connection using the Money Stream abstraction. Money Streams are bidirectional, meaning they can be used to send value from the Client to the Server or from the Server to the Client. A Connection may include any number of Money Streams.
+
+Multiple Money Streams MAY be used to separately account for value sent and received for different purposes. A Money Stream can be long-lived and the send and receive limits MAY be adjusted over time (for example, in response to an application-level action) or a Money Stream can be used to account for a single logical payment, in which case it would be closed when the payment is finished.
+
+Money Streams are identified by integers starting from 1. Odd-numbered Money Streams are initiated by the Client and even-numbered Money Streams are initiated by the Server. Money Stream identifiers are unrelated to Data Stream identifiers and a single Connection MAY have Money Streams and Data Streams with the same number.
+
+### Multiplexed Data Streams
+
+STREAM also offers the ability to send data through the Connection over ILP (rather than over the Internet). Data Streams are the abstraction used for grouping bytes sent over the connection. Data Streams are heavily inspired by [QUIC Bidirectional Streams](https://tools.ietf.org/html/draft-ietf-quic-transport-10#section-10).
+
+Data Streams are identified by integers starting from 1. Odd-numbered Data Streams are initiated by the Client and even-numbered Data Streams are initiated by the Server. Data Stream identifiers are unrelated to Money Stream identifiers and a single Connection MAY have Data Streams and Money Streams with the same number.
+
+### Stream and Connection Flow Control
+
+STREAM implements stream- and connection-level flow control. Clients and Servers communicate the maximum amount of money or data they are willing to receive on a given stream or for the connection as a whole. Limits may be changed and retransmitted throughout the life of the Connection to accept more money or data as desired by the application.
+
+### Connection Migration
+
+Clients and Servers MAY change the address the other entity will send packets to at any time by sending a `ConnectionNewAddress` frame with their new ILP address.
+
+**TODO**: Should we stipulate that the new ILP address is for an account in the same currency or a completely different one? (Should we forget the exchange rate we computed before if the address changes?)
```

Comment:
I think the exchange rate should be forgotten, otherwise there are tons of cases where a change in your moneyd or something would mess up open connections. The exchange rate shouldn't be too hard to recompute.